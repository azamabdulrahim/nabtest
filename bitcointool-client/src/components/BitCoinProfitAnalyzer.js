import React, { Component } from "react";
import BestProfitContainer from "./BestProfitContainer";
import HistoryContainer from "./HistoryContainer";

class BitCoinProfitAnalyzer extends Component {
  constructor() {
    super();

    this.state = {
      currency: "",
      date: undefined,
      refresh: false
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
    this.setState({ refresh: false });
  }

  onSubmit(e) {
    e.preventDefault();
    let { date } = this.state;
    if (!date) {
      alert("Please select date");
    } else {
      this.setState({ refresh: true });
    }
  }

  render() {
    let { refresh, date, currency } = this.state;
    let displayResult;
    if (refresh) {
      date = date.replace(/-/g, "");
      displayResult = (
        <div>
          <HistoryContainer date={date} currency={currency} refresh={refresh} />
          <BestProfitContainer
            date={date}
            currency={currency}
            refresh={refresh}
          />
        </div>
      );
    } else {
      displayResult = <div />;
    }

    return (
      <div>
        <div className="col-md-4 m-auto">
          <h1 className="text-center">Best Profit Calculator</h1>
          <hr />
          <form onSubmit={this.onSubmit}>
            <div className="form-group text-left">
              <label htmlFor="currency">Currency</label>
              <div>
                <select
                  id="currency"
                  name="currency"
                  className="form-control"
                  value={this.state.currency}
                  onChange={this.onChange}
                >
                  <option value="">All</option>
                  <option value="BTC">BTC</option>
                  <option value="ETC">ETC</option>
                  <option value="LTC">LTC</option>
                </select>
              </div>
            </div>
            <div className="form-group text-left">
              <label htmlFor="date">Date</label>
              <input
                id="date"
                type="date"
                className="form-control"
                name="date"
                value={this.state.date}
                onChange={this.onChange}
              />
            </div>

            <input type="submit" className="btn btn-primary" />
          </form>
          {displayResult}
        </div>
      </div>
    );
  }
}

export default BitCoinProfitAnalyzer;
