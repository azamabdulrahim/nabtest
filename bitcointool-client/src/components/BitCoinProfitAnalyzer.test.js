import React from "react";
import { shallow } from "enzyme";
import BitCoinProfitAnalyzer from "./BitCoinProfitAnalyzer";

describe("BitCoinProfitAnalyzer", () => {
  let wrapper;

  it("initializes correctly with no result shown", () => {
    wrapper = shallow(<BitCoinProfitAnalyzer />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("form").length).toEqual(1);
    expect(wrapper.find("#currency").length).toEqual(1);
    expect(wrapper.find("#date").length).toEqual(1);
    expect(wrapper.find("#results").length).toEqual(0);
  });

  it("updates component states when inputs change", () => {
    wrapper = shallow(<BitCoinProfitAnalyzer />);

    let refresh = wrapper.state().refresh;
    let currency = wrapper.state().currency;
    let date = wrapper.state().date;

    //test initial states
    expect(refresh).toBeFalsy();
    expect(currency).toEqual("");
    expect(date).toBeUndefined();

    wrapper
      .find("#currency")
      .simulate("change", { target: { name: "currency", value: "BTC" } });

    currency = wrapper.state().currency;
    expect(currency).toEqual("BTC");
    refresh = false; //reset
    refresh = wrapper.state().refresh;
    expect(refresh).toBeFalsy();

    wrapper
      .find("#currency")
      .simulate("change", { target: { name: "currency", value: "ETC" } });
    currency = wrapper.state().currency;
    expect(currency).toEqual("ETC");
    refresh = false; //reset
    refresh = wrapper.state().refresh;
    expect(refresh).toBeFalsy();

    wrapper
      .find("#date")
      .simulate("change", { target: { name: "date", value: "10-04-2019" } });

    date = wrapper.state().date;
    expect(date).toEqual("10-04-2019");
    refresh = false; //reset
    refresh = wrapper.state().refresh;
    expect(refresh).toBeFalsy();

    wrapper
      .find("#date")
      .simulate("change", { target: { name: "date", value: "11-04-2019" } });
    date = wrapper.state().date;
    expect(date).toEqual("11-04-2019");
    refresh = false; //reset
    refresh = wrapper.state().refresh;
    expect(refresh).toBeFalsy();
  });

  it("sets state.refresh to true and renders results when form is valid and submitted", () => {
    wrapper = shallow(<BitCoinProfitAnalyzer />);

    wrapper
      .find("#date")
      .simulate("change", { target: { name: "date", value: "10-04-2019" } });

    wrapper.find("form").simulate("submit", {
      preventDefault: () => {}
    });
    let refresh = wrapper.state().refresh;
    expect(refresh).toBeTruthy();

    wrapper.update();
    expect(wrapper.find("#results").length).toEqual(1);
  });
});
