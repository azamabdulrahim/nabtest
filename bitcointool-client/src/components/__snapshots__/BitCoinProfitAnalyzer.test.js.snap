// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BitCoinProfitAnalyzer initializes correctly with no result shown 1`] = `
<div>
  <div
    className="col-md-4 m-auto"
  >
    <h1
      className="text-center"
    >
      Best Profit Calculator
    </h1>
    <hr />
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group text-left"
      >
        <label
          htmlFor="currency"
        >
          Currency
        </label>
        <div>
          <select
            className="form-control"
            id="currency"
            name="currency"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              All
            </option>
            <option
              value="BTC"
            >
              BTC
            </option>
            <option
              value="ETC"
            >
              ETC
            </option>
            <option
              value="LTC"
            >
              LTC
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-group text-left"
      >
        <label
          htmlFor="date"
        >
          Date
        </label>
        <input
          className="form-control"
          id="date"
          name="date"
          onChange={[Function]}
          required={true}
          type="date"
        />
      </div>
      <input
        className="btn btn-primary"
        type="submit"
      />
    </form>
    <div />
  </div>
</div>
`;
